File: java/HttpClient/src/main/java/com/sdl/web/Auth/BasicHttpAuth.java
package com.sdl.web.Auth;

import com.sdl.web.Request.IHttpClientRequest;

public class BasicHttpAuth implements IAuthentication{

    public String Username;
    public String Password;

    public void setUsername(String username) {
        Username = username;
    }

    public String getUsername() {
        return Username;
    }

    public void setPassword(String password) {
        Password = password;
    }

    @Override
    public void ApplyManualAuthentication(IHttpClientRequest request) {

    }
}

File: net/src/Sdl.Tridion.Api.Client/HttpClient/Auth/BasicHttpAuth.cs
using System;
using System.Net;
using System.Text;
using Sdl.Tridion.Api.Http.Client.Request;

namespace Sdl.Tridion.Api.Http.Client.Auth
{
    public class BasicHttpAuth : IAuthentication
    {
        public string Username { get; set; }
        public string Password { get; set; }

        public ICredentials Credentials { get; } = null;

        public void ApplyManualAuthentication(IHttpClientRequest request)
        {
            // manually add authentication to request
            if (string.IsNullOrEmpty(Username) || string.IsNullOrEmpty(Password))
                return;
            // add basic http auth to request headers
            request.Headers.Add("Authorization", 
                $"Basic {Convert.ToBase64String(Encoding.Default.GetBytes($"{Username}:{Password}"))}");
        }
       
        public NetworkCredential GetCredential(Uri uri, string authType) => null;
    }
}
